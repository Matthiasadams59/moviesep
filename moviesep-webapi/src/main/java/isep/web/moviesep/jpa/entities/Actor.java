package isep.web.moviesep.jpa.entities;

import java.io.Serializable;
import javax.persistence.*;
import java.util.List;


/**
 * The persistent class for the actor database table.
 * 
 */
@Entity
@Table(name="actor")
@NamedQuery(name="Actor.findAll", query="SELECT a FROM Actor a")
public class Actor implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy=GenerationType.AUTO)
	@Column(name="actor_id", unique=true, nullable=false)
	private int actorId;

	@Column(name="first_name", nullable=false, length=45)
	private String firstName;

	@Column(nullable=false, length=20)
	private String gender;

	@Column(name="last_name", nullable=false, length=45)
	private String lastName;

	//bi-directional many-to-many association to Film
	@ManyToMany(mappedBy="actors")
	private List<Film> films;

	public Actor() {
	}

	public int getActorId() {
		return this.actorId;
	}

	public void setActorId(int actorId) {
		this.actorId = actorId;
	}

	public String getFirstName() {
		return this.firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	public String getGender() {
		return this.gender;
	}

	public void setGender(String gender) {
		this.gender = gender;
	}

	public String getLastName() {
		return this.lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName;
	}

	public List<Film> getFilms() {
		return this.films;
	}

	public void setFilms(List<Film> films) {
		this.films = films;
	}

}