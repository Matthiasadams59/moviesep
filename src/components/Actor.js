import React, {Component} from 'react'
import List from '@material-ui/core/List';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { getActor } from './../helpers/actor_helper';


class Actor extends Component {
    constructor(props) {
        super(props)

        this.state = {
            actorId: 0,
            lastName: '',
            firstName: '',
            gender: '',
            films: [] 
        }
    }


    componentDidMount= () => {
        getActor(this.props.match.params.id, (actor) => {
            this.setState({
                actorId: actor.actorId,
                lastName: actor.lastName,
                firstName: actor.firstName,
                gender: actor.gender,
                films: actor.films
            })
        })
    }

    render() {

        return(
            <div>
                <Typography variant="h2">ACTOR INFO</Typography>
                    <div>  
                        <List>
                            <ListItemText>
                                First Name: {this.state.firstName}
                            </ListItemText>
                            <ListItemText>
                                Last Name: {this.state.lastName}
                            </ListItemText>
                            <ListItemText>
                                Gender: {this.state.gender}
                            </ListItemText>
                            <Typography variant="h4">Movies</Typography>
                            <List>
                                {this.state.films.map((film, index) => {
                                    return (
                                        <ListItemText key={index}><Button href={"/film/" + film.filmId}>{film.title}</Button></ListItemText>
                                    )
                                })}
                            </List>
                        </List>
                    </div>
            </div>
        )
    }
}

export default Actor