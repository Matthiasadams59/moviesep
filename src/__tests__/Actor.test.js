import React from 'react';
import { shallow } from "enzyme";
import Actor from './../components/Actor';

describe("<Actor /> component", () => {
    let wrapper, component;

    const match = {
        params : { 
                    baseId : 1 //any id you want to set
                 }
       }

    beforeEach(() => {
        wrapper = shallow(<Actor match={match}/>);
        component = wrapper.instance();
    })

    it('renders correctly', () => {
        expect(wrapper).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();

    });

})


