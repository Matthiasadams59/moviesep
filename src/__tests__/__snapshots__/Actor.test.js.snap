// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Actor /> component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Actor
    match={
      Object {
        "params": Object {
          "baseId": 1,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          variant="h2"
        >
          ACTOR INFO
        </WithStyles(Typography)>,
        <div>
          <WithStyles(List)>
            <WithStyles(ListItemText)>
              First Name: 
              
            </WithStyles(ListItemText)>
            <WithStyles(ListItemText)>
              Last Name: 
              
            </WithStyles(ListItemText)>
            <WithStyles(ListItemText)>
              Gender: 
              
            </WithStyles(ListItemText)>
            <WithStyles(Typography)
              variant="h4"
            >
              Movies
            </WithStyles(Typography)>
            <WithStyles(List) />
          </WithStyles(List)>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "ACTOR INFO",
          "variant": "h2",
        },
        "ref": null,
        "rendered": "ACTOR INFO",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(List)>
            <WithStyles(ListItemText)>
              First Name: 
              
            </WithStyles(ListItemText)>
            <WithStyles(ListItemText)>
              Last Name: 
              
            </WithStyles(ListItemText)>
            <WithStyles(ListItemText)>
              Gender: 
              
            </WithStyles(ListItemText)>
            <WithStyles(Typography)
              variant="h4"
            >
              Movies
            </WithStyles(Typography)>
            <WithStyles(List) />
          </WithStyles(List)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(ListItemText)>
                First Name: 
                
              </WithStyles(ListItemText)>,
              <WithStyles(ListItemText)>
                Last Name: 
                
              </WithStyles(ListItemText)>,
              <WithStyles(ListItemText)>
                Gender: 
                
              </WithStyles(ListItemText)>,
              <WithStyles(Typography)
                variant="h4"
              >
                Movies
              </WithStyles(Typography)>,
              <WithStyles(List) />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "First Name: ",
                  "",
                ],
              },
              "ref": null,
              "rendered": Array [
                "First Name: ",
                "",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Last Name: ",
                  "",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Last Name: ",
                "",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Gender: ",
                  "",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Gender: ",
                "",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Movies",
                "variant": "h4",
              },
              "ref": null,
              "rendered": "Movies",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            variant="h2"
          >
            ACTOR INFO
          </WithStyles(Typography)>,
          <div>
            <WithStyles(List)>
              <WithStyles(ListItemText)>
                First Name: 
                
              </WithStyles(ListItemText)>
              <WithStyles(ListItemText)>
                Last Name: 
                
              </WithStyles(ListItemText)>
              <WithStyles(ListItemText)>
                Gender: 
                
              </WithStyles(ListItemText)>
              <WithStyles(Typography)
                variant="h4"
              >
                Movies
              </WithStyles(Typography)>
              <WithStyles(List) />
            </WithStyles(List)>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "ACTOR INFO",
            "variant": "h2",
          },
          "ref": null,
          "rendered": "ACTOR INFO",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(List)>
              <WithStyles(ListItemText)>
                First Name: 
                
              </WithStyles(ListItemText)>
              <WithStyles(ListItemText)>
                Last Name: 
                
              </WithStyles(ListItemText)>
              <WithStyles(ListItemText)>
                Gender: 
                
              </WithStyles(ListItemText)>
              <WithStyles(Typography)
                variant="h4"
              >
                Movies
              </WithStyles(Typography)>
              <WithStyles(List) />
            </WithStyles(List)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(ListItemText)>
                  First Name: 
                  
                </WithStyles(ListItemText)>,
                <WithStyles(ListItemText)>
                  Last Name: 
                  
                </WithStyles(ListItemText)>,
                <WithStyles(ListItemText)>
                  Gender: 
                  
                </WithStyles(ListItemText)>,
                <WithStyles(Typography)
                  variant="h4"
                >
                  Movies
                </WithStyles(Typography)>,
                <WithStyles(List) />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "First Name: ",
                    "",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "First Name: ",
                  "",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "Last Name: ",
                    "",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Last Name: ",
                  "",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "Gender: ",
                    "",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Gender: ",
                  "",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Movies",
                  "variant": "h4",
                },
                "ref": null,
                "rendered": "Movies",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
